homeassistant/components/smhi/weather.py

```diff
@@ -238,5 +258,11 @@ def forecast(self) -> list:
     @property
     def extra_state_attributes(self) -> dict:
         """Return SMHI specific attributes."""
-        if self.cloudiness:
-            return {ATTR_SMHI_CLOUDINESS: self.cloudiness}
+        extra_attributes = {}
+        if self.cloudiness is not None:
+            extra_attributes.update({ATTR_SMHI_CLOUDINESS: self.cloudiness})
```

Comment:
We don't need to create a dict and update the attribute dict. Just set the appropriate key in the attribute dict.
```suggestion
            extra_attributes[ATTR_SMHI_CLOUDINESS] = self.cloudiness
```

Reply:
Thank you, that is probably better.
I made a commit changing the three extra attributes as suggested.